routers/repo/migrate.go

```diff
@@ -25,6 +26,11 @@ const (
 
 // Migrate render migration of repository page
 func Migrate(ctx *context.Context) {
+	if setting.Repository.DisableMigrations {
+		ctx.ServerError("MigratePost", fmt.Errorf("cannot migrate; migrations disabled"))
```

Comment:
maybe it's not server error, just return 403 is enough

Reply:
Changed 500 to 403.